File: 示例代码/斐波那契.pl
# -*- coding: utf-8 -*-
# 测试于: ideone.com, perl6
# 本地测试perl v5.12.4: $ perl 斐波那契.pl

use utf8;

sub 斐波那契
{
    my ($数) = @_;
    return 0 if $数 == 0;
    return 1 if $数 == 1;
    return 斐波那契($数 - 1) + 斐波那契($数 - 2);
}

print &斐波那契(16)

File: 示例代码/斐波那契.rb
# -*- coding: utf-8 -*-
# 运行: $ ruby 斐波那契.rb

def 斐波那契(数值)
  return 数值 if 数值 < 2
  斐波那契(数值 - 1) + 斐波那契(数值 - 2)
end

def 调试输出(值)
  p 值
end

def 输出(值)
  print 值
end

输出 斐波那契(16)
